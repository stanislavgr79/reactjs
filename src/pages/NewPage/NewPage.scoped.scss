.new-page-container {
    font-size: 40px;
    color: #FFFFFF;
    font-weight: 100;
    padding-left: 12%;
    padding-top: 3%;
}