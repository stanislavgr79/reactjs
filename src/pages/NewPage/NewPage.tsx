import React from 'react';

import './NewPage.scoped.scss';
import { useLocation } from 'react-router-dom';

export default function NewPage(): JSX.Element {
  const location = useLocation();
  return (
    <div className="new-page">
      <div className="new-page-container">
        new page
        {location.pathname}
      </div>
      ;
    </div>
  );
}
