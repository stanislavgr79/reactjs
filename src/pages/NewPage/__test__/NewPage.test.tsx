import React from 'react';
import { render } from '@testing-library/react';
import ReactRouter from 'react-router';

import NewPage from '../NewPage';

describe('NewPage', () => {
  test('snapshot renders', () => {
    const mockLocation = {
      pathname: '/movies',
      search: '',
      state: null,
      hash: '',
    };

    jest.spyOn(ReactRouter, 'useLocation').mockReturnValue(mockLocation);
    const tree = render(<NewPage />);

    expect(tree.baseElement).toMatchSnapshot();
    expect(tree.baseElement).toContainHTML('<div class="new-page">');
  });
});
