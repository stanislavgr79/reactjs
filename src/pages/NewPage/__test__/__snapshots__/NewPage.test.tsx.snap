// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPage snapshot renders 1`] = `
<body>
  <div>
    <div
      class="new-page"
    >
      <div
        class="new-page-container"
      >
        new page
        /movies
      </div>
      ;
    </div>
  </div>
</body>
`;
